tests/conftest.py

```diff
@@ -29,41 +29,53 @@ def node(host, request):
         'mimic': 13,
         'dev': 99
     }
-    if not request.node.get_marker(node_type) and not request.node.get_marker('all'):  # noqa E501
-        pytest.skip("Not a valid test for node type: %s" % node_type)
 
-    if request.node.get_marker("no_lvm_scenario") and lvm_scenario:
+    # capture the initial/default state
+    valid_markers = []
```

Comment:
I know why you made this a list, but if we treated it like a boolean I think the logic would be easier to follow.

Why don't we rename this to something like ``run_test_on_node`` or ``test_is_applicable`` and default it to ``False``. Then below we just assign a ``True`` value to it if we find a marker that exists in group_names or if the marker is "all". This way we could avoid ``any()`` and just do something like ``if not run_test_on_node:`` on line 43 below.